### DISCUSSION TITLE
next-intl with app-router on azure

### DISCUSSION THREAD

**Discussion by LeopoldLerch** [amannn/next-intl#547] [Q&A]
Hi
I am currently building a next-js app and want to use the new app router. Everything works fine at my machine, however, not when deployed to  an azure web app. 
By using plain create-next-apps, I tracked it down to the middleware. It works with dynamic routing (using [locale]). I do not use the locales in the layout or page at all. so there everything default. 

As soon, as I a add the middleware-file with 
```
import createMiddleware from 'next-intl/middleware';

export default createMiddleware({
    // A list of all locales that are supported
    locales: ['en', 'de'],

    // If this locale is matched, pathnames work without a prefix (e.g. `/about`)
    defaultLocale: 'de'
});

export const config = {
    // Skip all paths that should not be internationalized. This example skips the
    // folders "api", "_next" and all files with an extension (e.g. favicon.ico)
    matcher: ['/((?!api|_next|.*\\..*).*)']
};
```

and upload the build, it is gone

Up from this I point only get 404 "Not found" pages from the next app. No difference if I set the locale in the url or not.
 (its the 404 from next). 

locally it is working absolutely fine.

Any Ideas are appreciated

**Comment by amannn** [comment]
I haven't heard about this one yet! Have you tried adding some logging to the web server, the middleware and components to isolate the issue? Would be helpful if you could share your findings!

**Comment by Francismori7** [comment]
Following this.
